File: tools/bench/polyglot/javascript/vec4_simd.js
const v = new Float32Array([1.0, 2.0, 3.0, 4.0]);
const s0 = 0.5, s1 = 0.25, s2 = 0.125, s3 = 0.0625;
for (let i = 0; i < 5000000; i++) {
    v[0] += s0;
    v[1] += s1;
    v[2] += s2;
    v[3] += s3;
}
const sum = v[0] + v[1] + v[2] + v[3];
console.log(Number(sum.toPrecision(6)).toString());

File: tools/bench/polyglot/python/vec4_simd.py
import struct

v = [1.0, 2.0, 3.0, 4.0]
s0, s1, s2, s3 = 0.5, 0.25, 0.125, 0.0625
for _ in range(5000000):
    v[0] += s0
    v[1] += s1
    v[2] += s2
    v[3] += s3

print(f"{sum(v):.6g}")
